### ISSUE TITLE
[apex] AvoidGlobalModifier false-negative with global method

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#2298]
**Affects PMD Version:** 6.22.0-SNAPSHOT

**Rule:** [AvoidGlobalModifier](https://pmd.github.io/latest/pmd_rules_apex_bestpractices.html#avoidglobalmodifier)

**Description:**

If just a method is declared as global, the rule doesn't detect this. The same is true for inner interfaces.
It seems, that the rule only detects global modifiers only on top-level types.

**Code Sample demonstrating the issue:**

Method:
``` java
class Foo {
   global Integer bar() {  // violation expected
   }
}
```
Inner Method:
``` java
class Foo {
    global interface Bar { // violation expected
    }
}
```

**Comment by rsoesemann** [comment]
The two examples are not valid apex. If a member is global the class has to be global as well.

**Comment by adangel** [comment]
Ok, thanks for the clarification.

**Comment by rsoesemann** [comment]
@oowekyala is this the issue you are referring to in Gitter? I dont use Gitter so it would be better if we can discuss Apex issues in Github issues.

**Comment by oowekyala** [comment]
@rsoesemann Yes that was it. This issue was very short-lived and honestly I had forgotten it was there and already resolved

> I dont use Gitter so it would be better if we can discuss Apex issues in Github issues.

Sure I'll keep it in mind
